PSPs/drafts/psp-34.md

```diff
@@ -0,0 +1,914 @@
+  "returnType": {
+    "displayName": [
+      "Option"
+    ],
+    "type": "Option<AccountId>"
+  },
+  "selector": "0x82db011d"
+}
+```
+
+##### **is_approved_for_all**(owner: AccountId, operator: AccountId) ➔ bool
+Selector: `0x53e20ef5` - first 4 bytes of `blake2b_256("PSP34::is_approved_for_all")`
+```json
+{
+  "args": [
+    {
+      "label": "owner",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    },
+    {
+      "label": "operator",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    }
+  ],
+  "docs": [
+    "Returns `true` if the operator is approved by the owner."
+  ],
+  "mutates": false,
+  "label": "PSP34::is_approved_for_all",
+  "payable": false,
+  "returnType": {
+    "displayName": [
+      "bool"
+    ],
+    "type": "bool"
+  },
+  "selector": "0x53e20ef5"
+}
+```
+
+##### **set_approval_for_all**(operator: AccountId, approved: bool) ➔ Result<(), PSP34Error>
+Selector: `0xf8938214` - first 4 bytes of `blake2b_256("PSP34::set_approval_for_all")`
+```json
+{
+  "args": [
+    {
+      "label": "operator",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    },
+    {
+      "label": "approved",
+      "type": {
+        "displayName": [
+        "bool"
+        ],
+        "type": "bool"
+      }
+    }
+  ],
+  "docs": [
+    "Approves or disapproves the operator for all tokens of the caller.",
+    "",
+    "On success a `ApprovalForAll` event is emitted.",
+    "",
+    "# Errors",
+    "",
+    "Returns `SelfApprove` error if it is self approve."
+  ],
+  "mutates": true,
+  "label": "PSP34::set_approval_for_all",
+  "payable": false,
+  "returnType": {
+    "displayName": [
+      "Result"
+    ],
+    "type": 1
+  },
+  "selector": "0xf8938214"
+}
+```
+
+##### **approve**(to: AccountId, id: Id) ➔ Result<(), PSP34Error>
+Selector: `0x1932a8b0` - first 4 bytes of `blake2b_256("PSP34::approve")`
+```json
+{
+  "args": [
+    {
+      "label": "to",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    },
+    {
+      "label": "id",
+      "type": {
+        "displayName": [
+           "Id"
+        ],
+        "type": "Id"
+      }
+    }
+  ],
+  "docs": [
+    "Approves the account to transfer the specified token on behalf of the caller.",
+    "",
+    "On success a `Approval` event is emitted.",
+    "",
+    "# Errors",
+    "",
+    "Returns `SelfApprove` error if it is self approve.",
+    "",
+    "Returns `NotApproved` error if caller is not owner of `id`."
+  ],
+  "mutates": true,
+  "label": "PSP34::approve",
+  "payable": false,
+  "returnType": {
+    "displayName": [
+      "Result"
+    ],
+    "type": 1
+  },
+  "selector": "0x1932a8b0"
+}
+```
+
+##### **transfer**(to: AccountId, id: Id, data: [u8]) ➔ Result<(), PSP34Error>
```

Comment:
I suggest using vec, as it would be easier to work with
```suggestion
##### **transfer**(to: AccountId, id: Id, data: Vec<u8>) ➔ Result<(), PSP34Error>
```